homeassistant/helpers/update_coordinator.py

```diff
@@ -214,20 +214,15 @@ def _schedule_refresh(self) -> None:
         # than the debouncer cooldown, this would cause the debounce to never be called
         self._async_unsub_refresh()
 
-        # We _floor_ utcnow to create a schedule on a rounded second,
-        # minimizing the time between the point and the real activation.
-        # That way we obtain a constant update frequency,
-        # as long as the update process takes less than 500ms
-        #
-        # We do not align everything to happen at microsecond 0
-        # since it increases the risk of a thundering herd
-        # when multiple coordinators are scheduled to update at the same time.
-        #
-        # https://github.com/home-assistant/core/issues/82231
-        self._unsub_refresh = event.async_track_point_in_utc_time(
+        # We use event.async_call_at because DataUpdateCoordinator does
+        # not guarantee an exact update interval
+        if self._next_refresh is None:
+            self._next_refresh = self.hass.loop.time()
+        self._next_refresh += self.update_interval.total_seconds()
```

Comment:
To avoid hammering slow to respond services, we could add some guard here such that if we're very late when scheduling, we reset `self._next_refresh` to `self.hass.loop.time()`.

Reply:
Maybe if we are at 3/4 of the interval reschedule ?